stock-filters/ChirpVillageBuilder.py

```diff
@@ -8,42 +8,28 @@
 
 
 from pymclevel import alphaMaterials, MCSchematic, MCLevel, BoundingBox
-from ChirpVillage.Buildings.BasicBuilding import BasicBuilding
+from Surface import Surface
+import utilityFunctions
+from random import randrange, uniform
+from ChirpVillage import BuildPlaneFinder, Builder
+import os
 from ChirpVillage.Biomes import BlockUtils
 from ChirpVillage.YardGenerator import YardGenerator
 
 inputs = (
     ('Build Chirp Village', 'label'),
-    ('Material1', alphaMaterials.StoneBricks),
     ('Creator: Chirp Nets', 'label'),
 )
-default_path = '/stock-filters/ChirpVillage/Buildings/'
-files = [
-    'ghast.schematic',
-]
 
 
 def perform(level, box, options):
+    surface = Surface(level, box)
```

Comment:
the surface is constructed in the YardGenerator.

Reply:
What was the reason for having the surface become apart of the yard generator again? Couldn't it just be passed in?

Reply:
It could be but its not valid until after the YardGenerator has at it

Reply:
actually i think i agree with you! john's path generator takes it as a constructor param too. I think it makes sense have the surface pass through the generators. did you want to make the change?